test/srv6-inject/src/nf.rs

```diff
@@ -90,36 +28,27 @@ struct MetaDataz {
 impl Default for MetaDataz {
     fn default() -> MetaDataz {
         MetaDataz {
-            next_header: NextHeader::NoNextHeader,
-            payload_len: 0,
             flow: FlowV6::default(),
             payload_diff: 0,
             segment_dst: Ipv6Addr::unspecified(),
         }
     }
 }
 
-#[inline]
-fn parse_meta<'a>(meta: &MetaDataz) -> (NextHeader, u16, FlowV6, i8, Segment) {
-    println!("Parsing MetaDATAZ {}", format!("{:?}", meta).green());
-    (
-        meta.next_header,
-        meta.payload_len,
-        meta.flow,
-        meta.payload_diff,
-        meta.segment_dst,
-    )
-}
-
 fn srh_into_packet(pkt: &mut Packet<Ipv6Header, MetaDataz>) -> () {
     let seg0 = Ipv6Addr::from_str("fe80::4").unwrap();
     let seg1 = Ipv6Addr::from_str("1ce:c01d:bee2:15:a5:900d:a5:11fe").unwrap();
     let segvec = vec![seg0, seg1];
-    srh_insert_packet_gen!(segvec,
-                         pkt,
-                         parse_meta(pkt.read_metadata()),
-                         Ipv6Header,
-                         [1 => U1, 2=> U2, 3 => U3, 4 => U4, 5 => U5, 6 => U6, 7 => U7, 8 => U8, 9 => U9, 10 => U10, 11 => U11, 12 => U12])
+    srh_insert!(segvec,
+                pkt,
+                (
+                    pkt.get_header().next_header(),
+                    segvec.len() as u8 - 1,
+                    0,
+                    0
+                ),
+                Ipv6Header,
+                [1 => U1, 2=> U2, 3 => U3, 4 => U4, 5 => U5, 6 => U6, 7 => U7, 8 => U8, 9 => U9, 10 => U10, 11 => U11, 12 => U12])
```

Comment:
in the future, can we just have this guy as a `static` or something, and not have to pass it in?

Reply:
yeah, can do that... I think.

Reply:
Yeah, looked into this,  this is not possible w/ the macro pattern inputs, as [blah => typeblah] is specific to macro syntax params. I'd have to move to a procedural macro, and I think we should do that at another time.